src/plots/cartesian/axes.js

```diff
@@ -514,6 +514,11 @@ axes.autoBin = function(data, ax, nbins, is2d, calendar) {
                 distinctData.minDiff / msexp, [0.9, 1.9, 4.9, 9.9], true);
         size0 = Math.max(minSize, 2 * Lib.stdev(data) /
             Math.pow(data.length, is2d ? 0.25 : 0.4));
+
+        // fallback if ax.d2c output BADNUMs
+        // e.g. when user try to plot categorical bins
+        // on a layout.xaxis.type: 'linear'
+        if(!isNumeric(size0)) size0 = 0;
```

Comment:
If a bin size of `0` made it through, this could lead to infinite loops or divide-by-zero errors somewhere else. It looks like `Axes.autoTicks` has its own fallback for this so `dummyAx.dtick` won't be zero, which is why this works for now, but perhaps it would be better to use `size0 = 1` instead so we don't have to depend on this behavior from `autoTicks`?